_playwright-tests/Integration/NodeJSLifecycleDates.spec.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  test,
+  expect,
+  RpmsApi,
+  ModuleStreamsApi,
+  ApiSearchRpmResponse,
+} from '../test-utils/_playwright-tests/test-utils/src';
+
+test.describe('Check NodeJS Lifecycle Dates', () => {
+  test('Verify RHEL9 nodejs modules have lifecycle dates', async ({ client }) => {
+    const repoUrl = 'https://cdn.redhat.com/content/dist/rhel9/9/aarch64/appstream/os/';
+
+    let repositoryRpmSearch: ApiSearchRpmResponse[];
+    await test.step('Search for nodejs in repository using include_package_sources', async () => {
+      repositoryRpmSearch = await new RpmsApi(client).searchRpm({
+        apiContentUnitSearchRequest: {
+          search: 'nodejs',
+          urls: [repoUrl],
+          includePackageSources: true,
+        },
+      });
+
+      expect(repositoryRpmSearch.length).toBeGreaterThan(0);
+
+      const nodejsPackage = repositoryRpmSearch.find((pkg) => pkg.packageName?.includes('nodejs'));
+      expect(nodejsPackage).toBeDefined();
+      expect(nodejsPackage?.packageName).toContain('nodejs');
+    });
+
+    await test.step('Verify nodejs has both package and module type streams with lifecycle dates', async () => {
+      const nodejsPackage = repositoryRpmSearch.find((pkg) => pkg.packageName?.includes('nodejs'));
+      expect(nodejsPackage?.packageSources).toBeDefined();
+      expect(nodejsPackage?.packageSources?.length).toBeGreaterThan(0);
+
+      const packageSources = nodejsPackage?.packageSources || [];
+
+      const packageTypeStream = packageSources.find((source) => source.type === 'package');
+      expect(packageTypeStream).toBeDefined();
+      expect(packageTypeStream?.type).toBe('package');
+
+      const moduleTypeStream = packageSources.find((source) => source.type === 'module');
+      expect(moduleTypeStream).toBeDefined();
+      expect(moduleTypeStream?.type).toBe('module');
+
+      expect(packageTypeStream?.startDate).toBeDefined();
+      expect(packageTypeStream?.endDate).toBeDefined();
+      expect(moduleTypeStream?.startDate).toBeDefined();
+      expect(moduleTypeStream?.endDate).toBeDefined();
+
+      expect(new Date(packageTypeStream?.startDate || '').getTime()).not.toBeNaN();
+      expect(new Date(packageTypeStream?.endDate || '').getTime()).not.toBeNaN();
+      expect(new Date(moduleTypeStream?.startDate || '').getTime()).not.toBeNaN();
+      expect(new Date(moduleTypeStream?.endDate || '').getTime()).not.toBeNaN();
```

Comment:
How often do these change? Can we predict/manage these values? This test makes is very hand-wavy and pretty much any mistake would not be caught (e.g. in Patch right now there is a bug that shows 1.1.1970 for updates, yet this test would be happy with that).

Reply:
Hi @marusak , I added an assert for a minimum start date. Waiting for @jlsherrill's  review now.

Reply:
As discussed during the standup, why not using exact dates? They won't change during lifetime of RHEL release and they would provide much stronger tests.

Reply:
there probably is some value in checking for a specific date, although IDK that our api would ever return a date improperly, and this wouldn't stop users of this api (IB) to display the date improperly if it was missing.  

These dates shouldn't change.  For the module stream one, you'd probably want to choose a specific version:
```
      const moduleTypeStream = packageSources.find((source) => source.type === 'module' && source.version = "19.0" );
```
as an example, not tested.

Reply:
OK, added that, thank you

Reply:
thats oaky, they can have different dates